src/sql/src/plan/query.rs

```diff
@@ -279,7 +279,15 @@ pub fn plan_root_query(
     scx: &StatementContext,
     mut query: Query<Raw>,
     lifetime: QueryLifetime,
-) -> Result<(HirRelationExpr, RelationDesc, RowSetFinishing), anyhow::Error> {
+) -> Result<
+    (
+        HirRelationExpr,
+        RelationDesc,
+        RowSetFinishing,
+        Vec<GlobalId>,
+    ),
```

Comment:
We were past due for extracting a struct here before, and now I think it's essential! Could you add a type like

```rust
struct PlannedQuery<E> {
    expr: E,
    desc: RelationDesc,
    finishing: RowSetFinishing,
    depends_on: Vec<GlobalId>,
}
```

and use that as the return value? That'll make the `depends_on` a good bit more self-documenting!